core/src/main/java/org/springframework/security/authorization/method/AuthorizationMethodInterceptors.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2002-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.authorization.method;
+
+import javax.annotation.security.DenyAll;
+import javax.annotation.security.PermitAll;
+import javax.annotation.security.RolesAllowed;
+
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.security.access.prepost.PostAuthorize;
+import org.springframework.security.access.prepost.PreAuthorize;
+
+/**
+ * A static factory for constructing common {@link AuthorizationMethodInterceptor}s
+ *
+ * @author Josh Cummings
+ * @since 5.6
+ * @see PreAuthorizeAuthorizationManager
+ * @see PostAuthorizeAuthorizationManager
+ * @see SecuredAuthorizationManager
+ * @see Jsr250AuthorizationManager
+ */
+public final class AuthorizationMethodInterceptors {
```

Comment:
I'm not clear on how this would work. Are you suggesting the following?

```java
public class AuthorizationManagerBeforeMethodInterceptor ... {
    private final Pointcut pointcut;
    private final AuthorizationManager<MethodInterceptor> manager;

    public AuthorizationManagerBeforeMethodInterceptor(AuthorizationManager<MethodInterceptor>, Class<? extends Annotation>... annotations) {
        this.pointcut = AuthorizationMethodPointcuts.forAnnotations(annotations);
        this.manager = manager;
    }

    // ...
}
```
I think this has the same accidental misconfiguration issue of:

```java
@Bean 
@Role(BeanDefinition.ROLE_INFRASTRUCTURE)
public PointcutAdvisor preAuthorize() {
    return new AuthorizationManagerBeforeMethodInterceptor(
        PostFilter.class, 
        new PreAuthorizeAuthorizationManager()
    );
}
```

It also seems to tie annotations directly to a general-purpose `AuthorizationManager` interceptor implementation.

Alternatively, we could introduce `PreAuthorizeBeforeMethodInterceptor` which would have a no-arg constructor and make its own decision about the `Pointcut` and `AuthorizationManager`.

Reply:
I think you would make constructor that took the same arguments as the static factory methods in `AuthorizationMethodInterceptors`